drivers/i2c/i2c_ll_stm32.c

```diff
@@ -43,26 +43,60 @@ LOG_MODULE_REGISTER(i2c_ll_stm32);
 #else
 #define STM32_I2C_DOMAIN_CLOCK_SUPPORT 0
 #endif
+int i2c_stm32_get_config(const struct device *dev, uint32_t *config)
+{
+	const struct i2c_stm32_config *cfg = dev->config;
+	struct i2c_stm32_data *data = dev->data;
+	struct i2c_config_timing *cfg_timings = (void *)config;
```

Comment:
I don't think you're supposed to return an other structure than `i2c_stm32_config`.
See i2c.h:
```
 * @param dev_config Pointer to return bit-packed 32-bit value of
 * the I2C controller configuration.
```
It would be logical that the structure to return is the same than the one provided in `i2c_configure`.
Aim is to allow application to compare current and expected configuration:
```
 * i2c_get_config can return cached config or probe hardware, but it has to be
 * up to date with current configuration.
```
This is not a big issue as timing will also be part of this structure

Reply:
When the the i2C node Device Tree has no <timings> property (i2c_stm32_get_config has noi2c_config_timing and  n_timings == 0) 
Should we return  "(I2C_MODE_CONTROLLER | speed)" but in that case there is no simple way to retrieve the I2C V2 <timing> 32bit value